src/xrpld/app/tx/detail/VaultWithdraw.cpp

```diff
@@ -50,35 +50,15 @@ VaultWithdraw::preclaim(PreclaimContext const& ctx)
     if (!assets.validNumber())
         return tecPRECISION_LOSS;
 
-    if (vaultAsset.native())
-        ;  // No special checks for XRP
-    else if (vaultAsset.holds<MPTIssue>())
-    {
-        auto mptID = vaultAsset.get<MPTIssue>().getMptID();
-        auto issuance = ctx.view.read(keylet::mptIssuance(mptID));
-        if (!issuance)
-            return tecOBJECT_NOT_FOUND;
-        if (!issuance->isFlag(lsfMPTCanTransfer))
-        {
-            // LCOV_EXCL_START
-            JLOG(ctx.j.error())
-                << "VaultWithdraw: vault assets are non-transferable.";
-            return tecNO_AUTH;
-            // LCOV_EXCL_STOP
-        }
-    }
-    else if (vaultAsset.holds<Issue>())
+    auto const& vaultAccount = vault->at(sfAccount);
+    auto const& account = ctx.tx[sfAccount];
+    auto const& dstAcct = ctx.tx[~sfDestination].value_or(account);
```

Comment:
You're taking a reference to a temporary here. Is that kosher across the board?

Reply:
Yes, these all fall under [lifetime extension of a temporary](https://en.cppreference.com/w/cpp/language/reference_initialization.html#Lifetime_of_a_temporary)